src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -409,10 +409,12 @@
     IsClauseGrammar=OneOf(
         "NULL",
         "NAN",
-        "NOTNULL",
-        "ISNULL",
         Ref("BooleanLiteralGrammar"),
     ),
+    # Define these as grammars to allow child dialects to disable them (since they are non-standard
+    # keywords)
+    IsNullGrammar=Ref.keyword("ISNULL"),
+    NotNullGrammar=Ref.keyword("NOTNULL"),
```

Comment:
Should these be `Nothing()` in ANSI if they are non-standard then?

My understanding is that `Postgres` is the only one to support these.

That would also allow you to remove some of the overrides below for Exasol, MySQL, Snowflake.

Reply:
I put these in here because the comments in the code say that `Expression_A_Grammar` was taken from the [Cockroach DB docs](https://www.cockroachlabs.com/docs/v20.2/sql-grammar.html#a_expr) and `ISNULL`/`NOTNULL` appear there.

That said, I wasn't entirely sure about this, so happy to change if preferred.

Reply:
Yeah I think better to reverse it if you don't mind. Cockroach DB isn't the same as ANSI and seems most dialects do not support this. But good to know you had a good reason for this!

Reply:
I've reworked it, so let me know what you think.